application/api/project.py

```diff
@@ -186,3 +186,54 @@ def get(self, session):
             return api_response(
                 True, APIMessages.SUCCESS, STATUS_OK,
                 {"projects_under_organization": projects_to_return})
+
+    @token_required
+    def delete(self, session):
+        """
+        DELETE call to delete project details.
+
+        Args:
+            session(object): User session
+            project_id: project_id to be deleted
+
+        Returns: Standard API Response with HTTP status code
+
+        """
+        db_connections=[]
+        suites=[]
+        user_associated=[]
+        get_project_parser = reqparse.RequestParser()
+        get_project_parser.add_argument(
+            'project_id', help=APIMessages.PARSER_MESSAGE,
+            required=True, type=int, location='args')
+        get_project_data = get_project_parser.parse_args()
+        project_obj = Project.query.filter_by(project_id =get_project_data['project_id'],is_deleted=False).first()
+        if not project_obj:
+            return api_response(False,
+                                    APIMessages.NO_RESOURCE.format('Project'),
+                                    STATUS_UNAUTHORIZED)
+        test_suite_obj=TestSuite.query.filter_by(project_id=project_obj.project_id).all()
+        db_connection_obj = DbConnection.query.filter_by(project_id = project_obj.project_id).all()
+        user_project_role_obj = UserProjectRole.query.filter_by(project_id=project_obj.project_id).all()
+        if not test_suite_obj and not db_connection_obj and not user_project_role_obj:
+            project_obj.is_deleted=True
+            project_obj.save_to_db()
+            
+            delete_message = APIMessages.DELETE_PROJECT_TRUE.format(project_obj.project_name)
+        else:
+            for each_obj in db_connection_obj:
+                db_connections.append({"db_connection_id":each_obj.db_connection_id,
+                "db_connection_name":each_obj.db_connection_name})
+            for each_suite in test_suite_obj:
+                suites.append({"suite_id":each_suite.test_suite_id, "suite_name":each_suite.test_suite_name})
+            for each_user in user_project_role_obj:
+                user_obj = User.query.filter_by(user_id=each_user.user_id).first()
```

Comment:
There will be multiple entries for the same user in user_project_role, handle it.